hadoop-project/pom.xml

```diff
@@ -2737,6 +2737,46 @@
         </dependencies>
       </dependencyManagement>
     </profile>
+    <profile>
+      <id>jdk17</id>
+      <activation>
+        <jdk>[17,)</jdk>
```

Comment:
Probably worth activating for Java 9+, where `maven.compiler.release` is supported.